app/models/experiment_statistics_generator.rb

```diff
@@ -0,0 +1,97 @@
+##
+# All methods are invoked with experiment and user as arguments
+class ExperimentStatisticsGenerator
+
+  # @param experiment [Experiment]
+  # @return [Hash] hash containing simulations statistics
+  def self.simulations_statistics(experiment, _=nil)
```

Comment:
Jak pisałem w opisie kodu, te metody są wywoływane przez send, każda dostaje te same parametry, ale tylko jedna wymaga usera. Nie podoba mi się to, dlatego prosiłem o propozycje zmian. Najlepsze wydaje mi się to, co proponuje @dkrol - instancjonowanie klasy z podaniem eksperymentu i usera w konstruktorze. To paskudztwo zniknie z kodu.